### ISSUE TITLE
Bug: when a full enemy harvester is deviated (switched sides) right before being picked up by carry-all, it will be deposited at enemy refinery and give credits to enemy player

### ISSUE THREAD

**Issue by stefanhendriks** [stefanhendriks/Dune-II---The-Maker#158]
When a full enemy harvester is deviated (switched sides) right before being picked up by carry-all, it will be deposited at enemy refinery and give credits to enemy player.

Scenario:

Given: Harvester of enemy
Harvester just reached max capacity
Harvester is being picked up by carry-all
Carry-all is not yet at Harvester
Harvester gets captured (deviator makes Harvester switch to other side)
Harvester gets picked up by (enemy!) Carry-All
Harvester gets deployed at enemy Refinery (and appears to be enemy harvester again with Refinery animation)
Harvester deposits credits for enemy.

Expected:
- when unit is captured, the enemy carry-all should not pick up unit

**Comment by itwhite81** [comment]
I think this is similar when an enemy unit is attacking and get's deviated.  It still "attacks" whatever it was attacking after deviation.  I would suggest on switching sides it needs to reset it's current mission state to none (I assume they are attack, move, harvest, request carryall)

**Comment by stefanhendriks** [comment]
@itwhite81 I agree, it should stop doing whatever they we're doing when deviated.
